// semantic-ui-vue/dist/commonjs/elements/Container/Container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiContainer',
  props: {
    fluid: {
      type: Boolean,
      description: 'Container has no maximum width.'
    },
    text: {
      type: Boolean,
      description: 'Reduce maximum width to more naturally accommodate text.'
    },
    textAlign: (0, _PropTypes.Enum)(['left', 'right', 'center', 'justified'], {
      description: 'Align container text.'
    })
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = (0, _lib.getElementType)(this);
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([(0, _lib.getChildProps)(this), {
        'class': (0, _lib.classes)('ui', this.textAlign, this.textAlign && this.textAlign !== 'justified' && 'aligned', this.text && 'text', this.fluid && 'fluid', 'container')
      }]),
      [this.$slots.default]
    );
  }
};
