// semantic-ui-vue/dist/commonjs/elements/Header/Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiHeader',
  props: {
    color: _PropTypes.Enum.Color,
    content: String,
    dividing: Boolean,
    floated: (0, _PropTypes.Enum)(['left', 'right']),
    icon: Boolean,
    image: Boolean,
    size: _PropTypes.Enum.Size,
    sub: Boolean
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = (0, _lib.getElementType)(this);
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([(0, _lib.getChildProps)(this), {
        'class': (0, _lib.classes)('ui', this.floated && this.floated + ' floated', this.color, this.size, this.dividing && 'dividing', this.icon && 'icon', this.image && 'image', this.sub && 'sub', 'header')
      }]),
      [this.$slots.default || this.content]
    );
  }
};
