// semantic-ui-vue/dist/commonjs/views/Item/ItemContent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function isValidAlignEnum(align) {
  _PropTypes.Enum.VerticalAlign.choices.includes(align);
}

exports.default = {
  name: 'SuiItemContent',
  description: 'An item can contain content',
  props: {
    verticalAlign: (0, _PropTypes.Enum)(['top', 'middle', 'bottom'], {
      description: 'Content can specify its vertical alignment (top, middle, bottom)'
    })
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = (0, _lib.getElementType)(this);
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([(0, _lib.getChildProps)(this), {
        'class': (0, _lib.classes)(isValidAlignEnum(this.verticalAlign) && this.verticalAlign + ' aligned', 'content')
      }]),
      [this.$slots.default]
    );
  },

  meta: {
    parent: 'SuiItem'
  }
};
