// semantic-ui-vue/dist/commonjs/views/Item/ItemGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _babelHelperVueJsxMergeProps = require('babel-helper-vue-jsx-merge-props');

var _babelHelperVueJsxMergeProps2 = _interopRequireDefault(_babelHelperVueJsxMergeProps);

var _lib = require('../../lib');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiItemGroup',
  description: 'A group of items',
  props: {
    unstackable: {
      type: Boolean,
      description: 'A table can specify how it stacks items responsively'
    },
    divided: {
      type: Boolean,
      description: 'Items can be divided to better distinguish between grouped content'
    },
    relaxed: {
      type: Boolean,
      description: 'A group of items can relax its padding to provide more negative space'
    },
    veryRelaxed: {
      type: Boolean,
      description: 'A group of items can very relax its padding to provide even more negative space'
    },
    link: {
      type: Boolean,
      description: 'An item can be formatted so that the entire contents link to another page'
    }
  },
  render: function render() {
    var h = arguments[0];

    var ElementType = (0, _lib.getElementType)(this);
    return h(
      ElementType,
      (0, _babelHelperVueJsxMergeProps2.default)([(0, _lib.getChildProps)(this), {
        'class': (0, _lib.classes)('ui', this.unstackable && 'unstackable', this.divided && 'divided', this.relaxed && !this.veryRelaxed && 'relaxed', this.veryRelaxed && 'very relaxed', this.link && 'link', 'items')
      }]),
      [this.$slots.default]
    );
  },

  meta: {
    parent: 'SuiItem'
  }
};
